src/borg/item.pyx

```diff
@@ -398,87 +402,90 @@ class ItemDiff:
 
         changes = []
 
-        if self.item1.is_link() or self.item2.is_link():
+        if self._item1.is_link() or self._item2.is_link():
             changes.append(self._link_string())
 
-        if 'chunks' in self.item1 and 'chunks' in self.item2:
+        if 'chunks' in self._item1 and 'chunks' in self._item2:
             changes.append(self._content_string())
 
-        if self.item1.is_dir() or self.item2.is_dir():
+        if self._item1.is_dir() or self._item2.is_dir():
             changes.append(self._dir_string())
 
-        if not (self.item1.get('deleted') or self.item2.get('deleted')):
+        if not (self._item1.get('deleted') or self._item2.get('deleted')):
             changes.append(self._owner_string())
             changes.append(self._mode_string())
 
         return ' '.join((x for x in changes if x))
 
     def _equal(self, chunk_iterator1, chunk_iterator2):
         # if both are deleted, there is nothing at path regardless of what was deleted
-        if self.item1.get('deleted') and self.item2.get('deleted'):
+        if self._item1.get('deleted') and self._item2.get('deleted'):
             return True
 
         attr_list = ['deleted', 'mode', 'source']
         attr_list += ['uid', 'gid'] if self._numeric_owner else ['user', 'group']
         for attr in attr_list:
-            if self.item1.get(attr) != self.item2.get(attr):
+            if self._item1.get(attr) != self._item2.get(attr):
                 return False
 
-        if 'mode' in self.item1:     # mode of item1 and item2 is equal
-            filemode = stat.filemode(self.item1.mode)
-            if filemode[0] == 'l' and 'source' in self.item1 and 'source' in self.item2 and self.item1.source != self.item2.source:
-                return False
+        if 'mode' in self._item1:     # mode of item1 and item2 is equal
+            filemode = stat.filemode(self._item1.mode)
+            try:
+                if self._item1.is_link() and self._item1.source != self._item2.source:
+                    return False
+            except AttributeError:
```

Comment:
guess you mean to catch AttributeErrors from `.source` here, but it would also catch them from `._itemN` or `.is_link` or from inside of the `.is_link` method.

maybe we can do it as in the original code. i think you micro-optimized here, but as is_link() is most often false, it does not make a big difference, I guess.